test/simulator_tests/birth_death_simulator_test.py

```diff
@@ -0,0 +1,419 @@
+import unittest
+
+import networkx as nx
+import numpy as np
+
+from typing import List, Tuple
+
+from cassiopeia.simulator.BirthDeathFitnessSimulator import (
+    BirthDeathFitnessSimulator,
+)
+from cassiopeia.simulator.TreeSimulator import (
+    TreeSimulatorError,
+)
+
+import cassiopeia.data.utilities as utilities
+
+
+def get_leaves(tree: nx.DiGraph) -> List[str]:
+    return [n for n in tree.nodes if tree.out_degree(n) == 0]
+
+
+def test_tree(tree: nx.DiGraph) -> Tuple[List[float], int, bool]:
+    """A helper function for testing simulated trees.
+
+    Outputs the (independently calculated) total lived time for each extant
+    lineage, the number of extant lineages, and whether the tree has the
+    expected node degrees (to ensure unifurcation collapsing was done
+    correctly).
+    """
+    tree = tree.copy()
+    tree.nodes["0"]["total_time"] = 0
+    for i in nx.edge_dfs(tree):
+        tree.nodes[i[1]]["total_time"] = (
+            tree.nodes[i[0]]["total_time"] + tree.edges[i]["length"]
+        )
+
+    leaves = get_leaves(tree)
+    times = []
+    for i in leaves:
+        times.append(tree.nodes[i]["total_time"])
+
+    out_degrees = [tree.out_degree(n) for n in tree.nodes]
+    out_degrees.pop(0)
+
+    correct_degrees = all(x == 2 or x == 0 for x in out_degrees)
+
+    return times, len(leaves), correct_degrees
+
+
+class BirthDeathSimulatorTest(unittest.TestCase):
```

Comment:
I think you can make a lot of the simulations below more deterministic with the addition of a numpy random state to the birth death simulator.

Reply:
Hi Matt,
I think some of these tests should work regardless of stochasticity, and so I want to ensure that. So therefore I've left the seed out on these, while others that require a seed I've included. Do you think this is alright?

Reply:
I think all our tests should be deterministic. If not, when a test fails, we might not be able to reproduce it.

Reply:
This sounds good I'll add some more seeds